sdk/python/tests/README.md

```diff
@@ -0,0 +1,341 @@
+
+### What are some important things to keep in mind when adding a new offline / online store?
+
+#### Type mapping/Inference
+
+Many problems arise when implementing your data store's type conversion to interface with Feast datatypes.
+1. You will need to correctly update `inference.py` so that Feast can infer your datasource schemas
+2. You also need to update `type_map.py` so that Feast knows how to convert your datastores types to Feast-recognized types in `feast/types.py`.
+
+#### Historical and online retrieval
+
+The most important functionality in Feast is historical and online retrieval. Most of the e2e and universal integration test test this functionality in some way. Making sure this functionality works also indirectly asserts that reading and writing from your datastore works as intended.
+
+
+### To include a new offline / online store in the main Feast repo
+
+* Extend `data_source_creator.py` for your offline store.
+* In `repo_configuration.py` add a new `IntegrationTestRepoConfig` or two (depending on how many online stores you want to test).
+    - Generally, you should only need to test against sqlite. However, if you need to test against a production online store, then you can also test against redis or dynamodb.
+* Run the full test suite with `make test-python-integration.`
+
+### Including a new offline / online store in the main Feast repo from external plugins with community maintainers.
+
+* This folder is for plugins that are officially maintained with community owners. Place the APIs in `feast/infra/offline_stores/contrib/`.
+* Extend `data_source_creator.py` for your offline store and implement the required APIs.
+* In `contrib_repo_configuration.py` add a new `IntegrationTestRepoConfig` (depending on how many online stores you want to test).
+* Run the test suite on the contrib test suite with `make test-python-contrib-universal`.
+
+### To include a new online store
+
+* In `repo_configuration.py` add a new config that maps to a serialized version of configuration you need in `feature_store.yaml` to setup the online store.
+* In `repo_configuration.py`, add new `IntegrationTestRepoConfig` for online stores you want to test.
+* Run the full test suite with `make test-python-integration`
+
+### To use custom data in a new test
+
+* Check `test_universal_types.py` for an example of how to do this.
+
+```python
+@pytest.mark.integration
+def your_test(environment: Environment):
+    df = #...#
+    data_source = environment.data_source_creator.create_data_source(
+        df,
+        destination_name=environment.feature_store.project
+    )
+    your_fv = driver_feature_view(data_source)
+    entity = driver(value_type=ValueType.UNKNOWN)
+    fs.apply([fv, entity])
+
+    # ... run test
+```
+
+### Running your own redis cluster for testing
```

Comment:
nit: redis should be Redis, here and elsewhere?

Reply:
fixed